src/App.js

```diff
@@ -1,61 +1,84 @@
 import React from 'react';
+import './App.css'
+import Header from './components/Header'
+import MainBody from './components/MainBody'
+import Footer from './components/Footer'
 
-function App() {
-  return (
-    <section className="todoapp">
-      <header className="header">
-        <h1>todos</h1>
+class App extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      todoItems: [],
+      userText: '',
+    }
+  };
 
-        <input
-          className="new-todo"
-          placeholder="What needs to be done?"
-          autoFocus=""
-        />
-      </header>
+  onTextFieldChange = (event) => {
+    const eValue = event.target.value;
+    this.setState((prev) => {
+        return {
+            ...prev, userText: eValue
+        }
+    })
+  };
+  onCheckboxChange = (number) => {
+    this.setState((prev) => {
+        const copyTodoArr = prev.todoItems;
+        copyTodoArr[number].completed = !copyTodoArr[number].completed;
+        return {
+            ...prev,
+            todoItems: [...copyTodoArr]
+        }
+    })
+  };
+
+  onSubmit = (e) => {
+      e.preventDefault();
+      this.setState((prev) => {
+          return {
+              todoItems: [
+                  {
+                      text: prev.userText,
```

Comment:
--- NEED TO HAVE ID

Add `id` field. It is mandatory to have ID and it should be unique for each item. Using `index` as `id` is not a good idea. This way you may face some issues working with each of them separately (e.g. toggle status on click).
Suggestion: use `Date.now()` as id
Suggestion 2: use `random string` library: https://www.npmjs.com/package/randomstring

Reply:
Thanks fot second suggestion! That's cool and very useful!